llvm/lib/Target/AMDGPU/AMDGPUCallLowering.cpp

```diff
@@ -296,6 +317,35 @@ bool AMDGPUCallLowering::canLowerReturn(MachineFunction &MF,
   return checkReturn(CCInfo, Outs, TLI.CCAssignFnForReturn(CallConv, IsVarArg));
 }
 
+/// Replace CallLowering::determineAndHandleAssignments() because we need to
+/// reserve ScratchRSrcReg when necessary.
+/// TODO: Investigate if reserving ScratchRSrcReg can be moved to calling conv
+/// functions. If so, then this function is not needed anymore -- we can just
+/// use CallLowering::determineAndHandleAssignments() as before.
+bool AMDGPUCallLowering::determineAndHandleAssignmentsLocal(
```

Comment:
Instead of creating determineAndHandleAssignmentsLocal, would be better to just add an overload of determineAndHandleAssignments in the base class that takes the CCInfo as an argument. IIRC the DAG has a similar split

Reply:
An overloaded function with a CCState argument has been created in the base class `CallLowering`.